src/js/lib/monster.ui.js

```diff
@@ -3870,11 +3870,12 @@ define(function(require) {
 			},
 			checkLength = function(event) {
 				var currentLenght = checkCurrentLength(),
-					allowedChars = currentLenght <= size ? size - currentLenght : 0;
+					allowedChars = currentLenght <= size ? size - currentLenght : -1,
+					allowedCharsText = allowedChars < 0 ? 0 : allowedChars;
 
-				allowedCharsLabel.text(allowedChars);
+				allowedCharsLabel.text(allowedCharsText);
 
-				if (allowedChars <= 0 || allowedChars > size) {
+				if (allowedChars < 0 || allowedChars > size) {
```

Comment:
Okay, so the problem is that we are using `allowedChars` for two different purposes:

- hold the remainder of chars available
- determine if current size exceeds max size.

To clarify the logic, we could use two different variables:

```js
const enforceMaxSizeFactory = maxSize => currentSize => ({
	remainder: Math.max(0, maxSize - currentSize),
	isGreaterThanMaxSize: currentSize > maxSize
});

const max = 100;
const enforceMaxSize = enforceMaxSizeFactory(max);

[99, 100, 101].forEach(
	curr => console.log({ curr, max, ...enforceMaxSize(curr) })
);

/*
{ curr: 99, max: 100, remainder: 1, isGreaterThanMaxSize: false }
{ curr: 100, max: 100, remainder: 0, isGreaterThanMaxSize: false }
{ curr: 101, max: 100, remainder: 0, isGreaterThanMaxSize: true }
*/
```
Which, transposed to our situation, would look like:
```js
checkLength = function(event) {
	var currentLength = checkCurrentLength(),
		isGreaterThanMaxSize = currentLength > size,
		allowedChars = Math.max(0, size - currentLength);

	allowedCharsLabel.text(allowedChars);

	if (isGreaterThanMaxSize) {
		// error handling
    }
```

Reply:
Thank you @azefiel :100: